plugins/modem-manager/fu-firehose-updater.c

```diff
@@ -845,9 +888,11 @@ fu_firehose_updater_class_init(FuFirehoseUpdaterClass *klass)
 }
 
 FuFirehoseUpdater *
-fu_firehose_updater_new(const gchar *port)
+fu_firehose_updater_new(const gchar *port, FuSaharaLoader *sahara)
 {
 	FuFirehoseUpdater *self = g_object_new(FU_TYPE_FIREHOSE_UPDATER, NULL);
-	self->port = g_strdup(port);
+	if (port != NULL)
+		self->port = g_strdup(port);
+	self->sahara = sahara;
```

Comment:
I think you want a g_object_ref here (and the unref in finalize too)

Reply:
fixed